test/xbgpu/test_engine.py

```diff
@@ -1452,3 +1573,150 @@ def get_stream_status(stream_name: str) -> bool:
+            for stopped_beam in stopped_beams:
+                stream_index = beam_outputs.index(stopped_beam)
+                beam_capture_stop_heap_indices[stream_index] = capture_stop_chunk_index * HEAPS_PER_FENGINE_PER_CHUNK
+                capture_stop_beams.append(("capture-stop", stopped_beam.name))
+            # NOTE: `get_free_chunk` is given `capture_stop_index` + 1 because
+            # `get_free_chunk` is actually called (just) before a :class:`bsend.Chunk`
+            # is sent. Ideally, we would wait for the n-th Chunk to send before
+            # issuing the `?capture-stop`, but both :meth:`bsend.Bsend.send_chunk`
+            # and :meth:`bsend.Chunk.send` are not asynchronous. As a result, we
+            # settled on :meth:`bsend.Bsend.get_free_chunk` in order to ensure the
+            # requests are at least complete before moving on with processing.
+            self._patch_method(
+                monkeypatch,
+                "get_free_chunk",
+                bsend.BSend,
+                capture_stop_chunk_index + 1,
+                client,
+                capture_stop_beams,
+            )
+
+        timestamp_step = n_samples_between_spectra * n_spectra_per_heap
+
+        corrprod_results, beam_results, _, _ = await self._send_data(
+            mock_recv_streams,
+            mock_send_stream,
+            corrprod_outputs,
+            beam_outputs,
+            heap_factory=lambda batch_index, present: self._default_heap_factory(
+                timestamp_step=timestamp_step,
+                n_ants=n_ants,
+                n_channels_per_substream=n_channels_per_substream,
+                n_spectra_per_heap=n_spectra_per_heap,
+                frequency=frequency,
+                data_value=10,
+                batch_index=batch_index,
+                present=present,
+            ),
+            present=np.ones((n_batches, n_ants), bool),
+            timestamp_step=timestamp_step,
+            n_channels_per_substream=n_channels_per_substream,
+            frequency=frequency,
+            n_spectra_per_heap=n_spectra_per_heap,
+            corrprod_capture_stop_heap_indices=corrprod_capture_stop_heap_indices,
+            beam_capture_stop_heap_indices=beam_capture_stop_heap_indices,
+        )
+
+        # NOTE: During receiving beam data, only data up to `beam_capture_stop_index`
+        # is stored. It is also verified that there is no more data in the stream.
+        # This is largely a sanity check that there is non-zero data for heaps that
+        # were transmitted before the `?capture-stop` request.
+        for beam_output, beam_capture_stop_heap_index in zip(beam_outputs, beam_capture_stop_heap_indices):
+            if beam_capture_stop_heap_index > 0:
+                np.testing.assert_equal(beam_results[beam_output.name][:beam_capture_stop_heap_index] != 0, True)
+            else:
+                np.testing.assert_equal(beam_results[beam_output.name] != 0, True)
```

Comment:
Since beam_results is now sized properly, can't this just be
```suggestion
            np.testing.assert_equal(beam_results[beam_output.name] != 0, True)
```